backend/remote/remote.go

```diff
@@ -168,29 +169,35 @@ func (rf *Factory) Create(address string) (types.Backend, error) {
 		return nil, err
 	}
 
-	go r.monitorPing(rpc)
+	go r.monitorPing(rpc, dataAddress /*use replica data addr for ping tracking*/)
 
 	return r, nil
 }
 
-func (r *Remote) monitorPing(client *rpc.Client) error {
+func (r *Remote) monitorPing(client *rpc.Client, targetAddr string) error {
```

Comment:
In fact targetAddr is information available with client right? You don't need the extra parameter here.